File: app/src/androidTest/java/com/thoughtworks/pafsilva/testautomationworkshop/utils/MockedServer.java
package com.thoughtworks.pafsilva.testautomationworkshop.utils;

import fi.iki.elonen.NanoHTTPD;

public class MockedServer extends NanoHTTPD {

    private final static int PORT = 4444;
    private String response;

    public MockedServer() {
        super(PORT);
    }

    @Override
    public Response serve(IHTTPSession session) {
        return new Response(response);
    }

    public void setResponse(String response) {
        this.response = response;
    }
}

File: app/src/androidTest/java/com/thoughtworks/pafsilva/testautomationworkshop/utils/MockedServer.kt
package com.thoughtworks.pafsilva.testautomationworkshop.utils

import fi.iki.elonen.NanoHTTPD

class MockedServer : NanoHTTPD(PORT) {

    companion object {
        private const val PORT = 4444
    }

    private var response: String? = null

    override fun serve(session: IHTTPSession): Response {
        return Response(response)
    }

    fun setResponse(response: String) {
        this.response = response
    }
}
